modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/fsm/CursorInitializationPhaseHandler.java

```diff
@@ -46,7 +47,13 @@ public Result handle(Query query) {
 
         CompletableFuture<Void> awaitFuture = query.executor.executePlan(context, plan)
                 .thenCompose(dataCursor -> {
-                    AsyncSqlCursor<InternalSqlRow> cursor = query.executor.createAndSaveSqlCursor(query, dataCursor); 
+                    AsyncSqlCursor<InternalSqlRow> cursor;
+                    try {
+                        cursor = query.executor.createAndSaveSqlCursor(query, dataCursor);
```

Comment:
As I see `createAndSaveSqlCursor` also used when processing batch of ddl statements from script (see `QueryExecutor.executeChildBatch`).
I'm not sure, but it seems there is a chance that the query could hang if an exception occurs in the same method.
Should we handle query cancellation exception there also?